### DISCUSSION TITLE
any plan to port nicegui to nimlang?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#1206] [Q&A]
https://github.com/nim-lang/Nim says
```
Nim is a statically typed compiled systems programming language. It combines successful concepts from mature languages like Python, Ada and Modula. Its design focuses on efficiency, expressiveness, and elegance (in that order of priority). Nim supports Windows, Linux, and Mac OS X
```

So if nicegui supports nimlang, then we can release a more compacted application

**Comment by falkoschindler** [comment]
Hi @retsyo,

As @rodja wrote in a similar discussion https://github.com/zauberzeug/nicegui/discussions/781#discussioncomment-5691516:
> Each new language would be an enormous task in itself. And some languages may not support the concepts which makes NiceGUI strong. But we are open for any contributions. Maybe someone wants to start a spike project for a specific language?

So it would probably be a standalone project replicating something _like_ NiceGUI. But it wouldn't be the same, since, e.g., there's no FastAPI in Nim. We would certainly need to re-invent parts of the wheel. 😉
